ext/oauth-client/providers/testoauth.com.json

```diff
@@ -0,0 +1,10 @@
+{
+  "title": "TestOAuth.com",
+  "class": "Civi\\OAuth\\TestOAuthDotComProvider",
+  "options": {
```

Comment:
Cool. :smile:  That looks so much easier than spinning-up a local oauth server and registering it (*which is what I've had to do before*)

Reply:
Yes, it's nice not to have to set up a whole 'nother thing! I'm wondering, what to you think about moving this provider to a separate tiny extension within the oauth-client directory? I could also include the search afform that I describe in the testing steps. That way we keep test code separate.

Reply:
@highfalutin - (1) I think it's OK in `oauth-client`. 

(2) General feeling: OAuth is a fiddly architecture (*what with providers' myriad processes+rules for client-registration, scopes, metadata, and small differences of interpretation*). The fiddliness means that, basically, anyone with `oauth-client` is liable to need some debugging tools.

(3) That search/afform is very similar to the backend ["Inspect" action](https://github.com/civicrm/civicrm-core/assets/1336047/5458fa46-e4c4-479c-8638-dcc756c821b9) (`oauthJwtDebug`), which is already in `oauth-client`.  It relies on a permission-guard (`manage oauth client secrets`). I think it makes sense to have similar policy/tooling for `OAuthSysToken` / `OAuthContactToken` / `OAuthSessionToken`. Alas, `oauthJwtDebug` regressed at some point. ☹️ (*Work-around: replace [routeParams.id](https://github.com/colemanw/civicrm-core/blob/ff3347feddc4c514012495cdd837e09b057aec37/ext/oauth-client/ang/oauthJwtDebug.aff.html#L8C87-L8C101) with the actual number...*)

(4) So I think your approach with SK+AF is more contemporary, and it would be good to incorporate something like it. But perhaps....

* Instead of querying `OAuthSessionToken`, query the `EntitySet` union of `OAuthSysToken` / `OAuthContactToken` / `OAuthSessionToken`.
* Incorporate [`{{token.access_token|unvalidatedJwtDecode|json}}`](https://github.com/totten/civicrm-core/blob/ff3347feddc4c514012495cdd837e09b057aec37/ext/oauth-client/ang/oauthJwtDebug.aff.html#L28)
* Update the existing "Inspect" link
* Ensure it abides the permission `manage oauth client secrets`